src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -318,10 +318,10 @@
                               (format "Attempt to set url-prefix to %s when it's already been set to %s" url-prefix @old-url-prefix)))))
   (shared-globals [this]
                   (:shared-globals (service-context this)))
-  (query [this query-obj version query-expr paging-options row-callback-fn]
-         (let [{db :scf-read-db} (get (service-context this) :shared-globals)
-               url-prefix @(get (service-context this) :url-prefix)]
-           (qeng/stream-query-result query-obj version query-expr paging-options db url-prefix row-callback-fn))))
+  (query [this version query-expr paging-options row-callback-fn]
+         (let [query-options (-> (select-keys (get (service-context this) :shared-globals) [:scf-read-db :warn-experimental])
```

Comment:
I might pull out the `(get (service-context this) :shared-gloabls)` to be the first argument of the `->`